perf/load/pilot/load_test.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# this program checks the config push latency for the pilot.
+import sys
+import os
+import time
+import typing
+import subprocess
+
+cwd = os.getcwd()
+path = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '../../../metrics'))
+sys.path.insert(0, path)
+
+from prometheus import Query, Alarm, Prometheus
+import check_metrics
+
+
+def setup_promethus():
+    port = os.environ.get("PROM_PORT", "9990")
+    namespace, deployment = check_metrics.find_prometheus()
+    port_forward = subprocess.Popen([
+        'kubectl',
+        '-n', namespace,
+        'port-forward',
+        deployment,
+        '%s:9090' % port
+    ], stdout=subprocess.PIPE)
+    port_forward.stdout.readline()  # Wait for port forward to be ready
+    return Prometheus('http://localhost:%s/' % port)
+
+
+def envoy_cds_version_count(prom: Prometheus):
+    return prom.fetch_value('count(count_values("value", envoy_cluster_manager_cds_version))')
+
+
+def setup_pilot_loadtest(instance, svc_entry: int):
+    helm = 'serviceEntries=%d,instances=%d' % (svc_entry, instance)
+    env = os.environ
+    env['HELM_FLAGS'] = helm
+    p = subprocess.Popen([
+        './setup.sh',
+    ], env=env)
+    p.wait()
+
+
+def wait_till_converge(prom: Prometheus):
+    while True:
+        count = envoy_cds_version_count(prom)
+        if count == 1:
+            print('envoy version converged, returning')
+            return
+        print('envoy version not converged yet, count = %s' % count)
```

Comment:
Does the `pilot_proxy_convergence_time` give us the same thing? It sounds the same but I have never looked at much

Reply:
i looked at pilot code, and here's my understanding.
- `pilot_proxy_convergence_time` was registered in pilot, measuring xds push duration.
- this metric, per @mandarjog told me, is from envoy sidecar, reporting the version of the cds.

So the second one is closer to the e2e config to push latency, the first one exclude the time pilot get notified, xDS construction computation time.

Reply:
Makes sense to me

Reply:
`envoy_cluster_manager_cds_version` and the same one with lds does not work well when sidecar resource is enabled. I think we have talked about it.

Reply:
not sure if i get it. this is the metrics you recommended. the setup currently does not use sidecar.

Reply:
i guess you mean when sidecar is used, we don't expect all workloads' xDS version to be the same?